### ISSUE TITLE
generate-event sns adds in &quot; html entities

### ISSUE THREAD

**Issue by ste00martin** [awslabs/aws-sam-cli#755]
On aws-cli version 0.6.2 

Expected output of `sam local generate-event sns notification --message "$(cat event.json)"`

```
{
    "Records": [
        {
            "EventVersion": "1.0",
            "EventSource": "aws:sns",
            "EventSubscriptionArn": "arn:aws:sns:EXAMPLE",
            "Sns": {
                "MessageId": "95df01b4-ee98-5cb9-9903-4c221d41eb5e",
                "Signature": "EXAMPLE",
                "Type": "Notification",
                "TopicArn": "arn:aws:sns:us-east-1:111122223333:ExampleTopic",
                "MessageAttributes": {
                    "Test": {
                        "Type": "String",
                        "Value": "TestString"
                    },
                    "TestBinary": {
                        "Type": "Binary",
                        "Value": "TestBinary"
                    }
                },
                "SignatureVersion": "1",
                "Timestamp": "1970-01-01T00:00:00.000Z",
                "SigningCertUrl": "EXAMPLE",
                "Message": "{\n  \"foo\": \"bar\"\n}",
                "UnsubscribeUrl": "EXAMPLE",
                "Subject": "example subject"
            }
        }
    ]
}
```
Actual output:
```
{
  "Records": [
    {
      "EventVersion": "1.0",
      "EventSubscriptionArn": "arn:aws:sns:us-east-1::ExampleTopic",
      "EventSource": "aws:sns",
      "Sns": {
        "SignatureVersion": "1",
        "Timestamp": "1970-01-01T00:00:00.000Z",
        "Signature": "EXAMPLE",
        "SigningCertUrl": "EXAMPLE",
        "MessageId": "95df01b4-ee98-5cb9-9903-4c221d41eb5e",
        "Message": "{
  &quot;foo&quot;: &quot;bar&quot;
}",
        "MessageAttributes": {
          "Test": {
            "Type": "String",
            "Value": "TestString"
          },
          "TestBinary": {
            "Type": "Binary",
            "Value": "TestBinary"
          }
        },
        "Type": "Notification",
        "UnsubscribeUrl": "EXAMPLE",
        "TopicArn": "arn:aws:sns:us-east-1:123456789012:ExampleTopic",
        "Subject": "example subject"
      }
    }
  ]
}
```
My event.json file:
```
{
  "foo": "bar"
}
```

On aws-cli version 0.3.0 it works correctly. However, the way you have to invoke generate-event is different between versions. The way to call generate-event on v0.3.0 with the correctly formatted response is this way: `sam local generate-event sns --message "$(cat event.json)"` which outputs the first json blob.

For now my workaround is to use the older version; but when will this be fixed going forward so:
a) newer versions of aws-cli has backwards compatibility
b) json files are parsed correctly
c) you can pass in a json file directly into the `generate-event` command without the `cat` command?

**Comment by jackdbernier** [comment]
It looks like https://github.com/awslabs/aws-sam-cli/commit/dcd65f4442233d054c5ee8371e0c5e5e2f7ed7b4 is the commit that introduced this issue. With this commit, sam uses a templating engine. The solution would be to make a PR and change `{{message}}` to `{{{message}}}` in this file https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/local/lib/generated_sample_events/events/sns/Sns.json#L12 (and ideally doing the same for all events).

You can look at https://github.com/noahmorrison/chevron/blob/master/tests/test.mustache for more details about how the templating engine renders variables.

**Comment by jfuss** [comment]
@ste00martin Thanks for the report. @jackdbernier is correct. `{{message}}` will be html echoed by default and that using `{{{message}}}` instead is the correct behavior we want across all events. I have a patch for [this in my fork](https://github.com/jfuss/aws-sam-cli/tree/non-html-encode-events). Hoping to get this out for a PR after I make a second pass through all the events.

More details on how mustache behaves in the variable section [here](http://mustache.github.io/mustache.5.html)

**Comment by jackdbernier** [comment]
The combo of triple braces and json escaping works.
For instance,
```
sam local generate-event sns notification --message "$(jq '.|tojson' event.json)"
```
will output 
```
{
  "Records": [
    {
      "EventSource": "aws:sns",
      "EventVersion": "1.0",
      "EventSubscriptionArn": "arn:aws:sns:us-east-1::ExampleTopic",
      "Sns": {
        "Type": "Notification",
        "MessageId": "95df01b4-ee98-5cb9-9903-4c221d41eb5e",
        "TopicArn": "arn:aws:sns:us-east-1:123456789012:ExampleTopic",
        "Subject": "example subject",
        "Message": ""{\"name\":\"Jacques\"}"",
        "Timestamp": "1970-01-01T00:00:00.000Z",
        "SignatureVersion": "1",
        "Signature": "EXAMPLE",
        "SigningCertUrl": "EXAMPLE",
        "UnsubscribeUrl": "EXAMPLE",
        "MessageAttributes": {
          "Test": {
            "Type": "String",
            "Value": "TestString"
          },
          "TestBinary": {
            "Type": "Binary",
            "Value": "TestBinary"
          }
        }
      }
    }
  ]
}
```

if you omit the jq part you will end up with something like `"Message": "{"name": "Jacques"}",` and that is not valid. It used to work before and I think it is because it was using https://docs.python.org/3/library/json.html to generate the event. So `--message "$(cat event.json)"` needs to become something like `--message "$(jq '.|tojson' event.json)"`. This is not as straightforward as it used to be but with 1 or 2 examples. It will definitely work out just fine.

Thanks for jumping on that.

**Comment by jfuss** [comment]
This was released in [v0.9.0](https://github.com/awslabs/aws-sam-cli/releases/tag/v0.9.0)

Closing
